File: Location/app/src/main/java/com/example/location/MainActivity.java
package com.example.location;

import androidx.appcompat.app.AppCompatActivity;

import android.content.BroadcastReceiver;
import android.content.Intent;
import android.content.IntentFilter;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    BroadcastReceiver broadcast;
    TextView textView;

    public static final String SHARED_PREFS = "sharedPrefs";
    public static final String TEXT1 = "text";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

//        broadcast = new MyReceiver();
        textView = findViewById(R.id.texter);

        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE);
        String string = sharedPreferences.getString(TEXT1, "");
        textView.setText(string);

    }

//    @Override
//    protected void onStart() {
//        super.onStart();
//        IntentFilter intentFilter = new IntentFilter("android.location.PROVIDERS_CHANGED");
//        intentFilter.addCategory(Intent.ACTION_PROVIDER_CHANGED);
//        registerReceiver(broadcast,intentFilter);
//    }
}

File: FirebaseNotifications-master/app/src/main/java/com/androiddevs/firebasenotifications/MainActivity.kt
package com.androiddevs.firebasenotifications

import android.content.Context
import android.media.MediaPlayer
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.widget.Toast
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.messaging.FirebaseMessaging
import com.google.gson.Gson
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

const val TOPIC = "/topics/myTopic2"

private lateinit var mediaPlayer: MediaPlayer
private lateinit var runnable:Runnable
private var handler: Handler = Handler()
private var pause:Boolean = false

class MainActivity : AppCompatActivity() {

    val TAG = "MainActivity"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        FirebaseService.sharedPref = getSharedPreferences("sharedPref", Context.MODE_PRIVATE)
        FirebaseInstanceId.getInstance().instanceId.addOnSuccessListener {
            FirebaseService.token = it.token
            etToken.setText(it.token)
        }
        FirebaseMessaging.getInstance().subscribeToTopic(TOPIC)

        SirenStop.setOnClickListener{
            mediaPlayer = MediaPlayer.create(applicationContext,R.raw.alarm_safety)
            mediaPlayer.start()
            Toast.makeText(this, "media playing", Toast.LENGTH_SHORT).show()
//            if(mediaPlayer.isPlaying || pause.equals(true)){
//                pause = false
//                mediaPlayer.stop()
//                mediaPlayer.reset()
//                mediaPlayer.release()
//                handler.removeCallbacks(runnable)
//
//                Toast.makeText(this,"media stop",Toast.LENGTH_SHORT).show()
//            }
        }

        btnSend.setOnClickListener {
            val title = etTitle.text.toString()
            val message = etMessage.text.toString()
            val recipientToken = etToken.text.toString()
            if(title.isNotEmpty() && message.isNotEmpty() && recipientToken.isNotEmpty()) {
                PushNotification(
                        NotificationData(title, message),
                        recipientToken
                ).also {
                    sendNotification(it)
                }
            }
        }
    }

    private fun sendNotification(notification: PushNotification) = CoroutineScope(Dispatchers.IO).launch {
        try {
            val response = RetrofitInstance.api.postNotification(notification)
            if(response.isSuccessful) {
                Log.d(TAG, "Response: ${Gson().toJson(response)}")
            } else {
                Log.e(TAG, response.errorBody().toString())
            }
        } catch(e: Exception) {
            Log.e(TAG, e.toString())
        }
    }
}